cmd/tracker/resume_bundle.go

```
@@ -0,0 +1,41 @@
+//
+// The empty-vs-empty case (resume a .dip-started run against a .dip) is the
+// only no-identity-change case that's silently allowed; it preserves existing
+// behavior for plain .dip workflows.
+func verifyResumeBundle(checkpointIdentity, currentIdentity string, force bool) error {
+	if checkpointIdentity == currentIdentity {
+		return nil
+	}
+	if force {
+		return nil
+	}
+	return fmt.Errorf("%w\n  run was started against: %s\n  current bundle:          %s\nThe pipeline source has changed since this run was started. To resume against a different bundle, pass --force-bundle-mismatch.",
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not end with punctuation or newlines

(staticcheck)